include/vkwrap/vkwrap/mman.h

```diff
@@ -0,0 +1,533 @@
+    using vk::PipelineStageFlags;
+
+    PipelineStageFlags src{ PipelineStageFlagBits::eNone };
+    PipelineStageFlags dst{ PipelineStageFlagBits::eNone };
+
+    if ( old_layout == ImageLayout::eUndefined && new_layout == ImageLayout::eTransferDstOptimal )
+    {
+        src = PipelineStageFlagBits::eTopOfPipe;
+        dst = PipelineStageFlagBits::eTransfer;
+    } else if ( old_layout == ImageLayout::eTransferDstOptimal && new_layout == ImageLayout::eShaderReadOnlyOptimal )
+    {
+        src = PipelineStageFlagBits::eTransfer;
+        dst = PipelineStageFlagBits::eFragmentShader;
+    } else if ( old_layout == ImageLayout::eUndefined && new_layout == ImageLayout::eDepthStencilAttachmentOptimal )
+    {
+        src = PipelineStageFlagBits::eTopOfPipe;
+        dst = PipelineStageFlagBits::eEarlyFragmentTests;
+    } else
+    {
+        throw Error{ "choosePipelineStages: unimplemented layout transition." };
+    }
+
+    return { src, dst };
+} // choosePipelineStages
+
+class Mman
+{
+
+  private:
+    struct BufferInfo
+    {
+        VmaAllocation allocation;
+        vk::DeviceSize size;
+    }; // struct BufferInfo
+
+    struct ImageInfo
+    {
+        VmaAllocation allocation;
+        vk::Format format;
+        vk::ImageLayout layout;
+        vk::Extent3D extent;
+    }; // struct ImageInfo
+
+    class OneTimeCommand : private vk::UniqueCommandBuffer
+    {
+
+      private:
+        using Base = vk::UniqueCommandBuffer;
+
+      private:
+        vk::Queue m_queue;
+
+        vk::UniqueCommandBuffer createCommandBuffer( vk::Device device, vk::CommandPool cmd_pool ) const
+        {
+            vk::CommandBufferAllocateInfo alloc_info{
+                .commandPool = cmd_pool,
+                .level = vk::CommandBufferLevel::ePrimary,
+                .commandBufferCount = 1 };
+
+            return std::move( device.allocateCommandBuffersUnique( alloc_info )[ 0 ] );
+        } // createCommandBuffer
+
+        void begin() const
+        {
+            // clang-format off
+            vk::CommandBufferBeginInfo begin_info{ 
+                .flags = vk::CommandBufferUsageFlagBits::eOneTimeSubmit };
+            // clang-format on
+
+            Base::get().begin( begin_info );
+        } // begin
+
+        void end() const { Base::get().end(); }
+
+        void submit() const
+        {
+            vk::SubmitInfo submit_info{ .commandBufferCount = 1, .pCommandBuffers = &Base::get() };
+
+            m_queue.submit( std::array{ submit_info } );
+        } // submit
+
+        void wait() const { m_queue.waitIdle(); }
+
+      public:
+        OneTimeCommand( vk::Device device, vk::CommandPool cmd_pool, vk::Queue queue )
+            : Base{ createCommandBuffer( device, cmd_pool ) },
+              m_queue{ queue }
+        {
+        } // oneTimeCommand
+
+        void submitAndWait( std::function<void( vk::CommandBuffer& )> func )
+        {
+            begin();
+            func( this->get() );
```

Comment:
What does ```this->``` do here? Previously you used ```Base::```, which is more clear

Reply:
Yep, `Base::` is more clear.